util/movement-types/src/lib.rs

```diff
@@ -1,11 +1,21 @@
 use aptos_types::state_proof::StateProof;
 use core::fmt;
 use serde::{Deserialize, Serialize};
+use std::cmp::Ordering;
+use std::collections::BTreeSet;
 
 #[derive(Serialize, Deserialize, Clone, Default, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
-pub struct Id(pub [u8; 32]);
```

Comment:
I think it's good to derive `Copy` as well. We can commit to `Id` being of fixed size, so we don't need to gum the code with clone calls.

Reply:
See questions above about whether we want a shared base `Id`. I'd wonder about reuse and extensions of the Id.